### PR TITLE
Disable cache reclaim when cgroup v2.

### LINKED ISSUE #8497: Make cache reclaim support v2

### PR DESCRIPTION
**Pull request by utam0k**
## Description

Because cache reclaim supports only when using cgroup v1.

## Related Issue(s)

Fixes #8497

## How to test

1. Prepare an environment with cgroup v2 enabled e.g. `gitpod-k3s-202202181722`
2. Start a workspace
3. Confirm that ws-daemon doesn't complain error message about the cache reclaim

## Release Notes

```release-note
Disable cache reclaim when cgroup v2
```

## Documentation

No

### CODE REVIEW

**Comment by utam0k** [comment]
This PR is a draft because this pr is waiting for solving this discussion.
https://github.com/gitpod-io/gitpod/pull/8471#discussion_r819202598

File: components/ws-daemon/pkg/daemon/daemon.go [hunk 1]
```diff
@@ -52,12 +53,22 @@ func NewDaemon(config Config, reg prometheus.Registerer) (*Daemon, error) {
...
 		cgCustomizer,
 		markUnmountFallback,
-	)
+	}
+
+	unified, err := cgroups.IsUnifiedCgroupSetup()
+	if err != nil {
+		return nil, xerrors.Errorf("could not determine cgroup setup: %w", err)
+	}
+	if unified {
```
Comment (Furisto) [inline components/ws-daemon/pkg/daemon/daemon.go:67]: Shouldn't this be the other way around? That is add cache reclaim if we are on a system with a legacy (v1) cgroup setup?

Comment (utam0k) [reply components/ws-daemon/pkg/daemon/daemon.go:67]: you are right! Thanks.

**Comment by utam0k** [comment]
@Furisto PTAL

**Review by Furisto — APPROVED** [review]

**Pull request merged**